src/test/java/de/tum/in/www1/artemis/FileUploadSubmissionIntegrationTest.java

```diff
@@ -109,6 +109,15 @@ public void submitFileUploadSubmission() throws Exception {
         checkDetailsHidden(returnedSubmission, true);
     }
 
+    @Test
+    @WithMockUser(value = "student3")
+    public void submitFileUploadSubmission_emptyFileContent() throws Exception {
```

Comment:
Just a couple of ideas for additional tests:
Do we also have a test checking if the file is actually stored in the filesystem? Especially with the new hash comparison, we could additionally check if the stored file has the same hash. 
Also, another idea: Mock the saveFileForSubmission method and store the wrong file forcing the IOException caused by different hashes.

Reply:
I now check that the file is actually stored in the `submitFileUploadSubmission` test case.
I also thought about mocking the `saveFileForSubmission` method, but I didn't find any way to mock private method and I didn't want to make it public only for testing purposes.